File: fizzbuzz/fizzbuzz.cpp
#include <iostream>

int main() {
  for (int i = 1; i <= 100; i++) {
    if (i % 3 == 0 && i % 5 == 0) {
      std::cout << "fizzbuzz\n";
    } else if (i % 3 == 0) {
      std::cout << "fizz\n";
    } else if (i % 5 == 0) {
      std::cout << "buzz\n";
    } else {
      std::cout << i << "\n";
    }
  }

  return 0;
}

// how to run:
// g++ fizzbuzz.cpp -o fizzbuzz-cpp ; ./fizzbuzz-cpp

File: fizzbuzz/fizzbuzz.rs
//! Rust implementation of FizzBuzz
//! source: https://www.fotonixx.com/posts/rust-fizzbuzz/

fn main() {
    for x in 1..=100 {
        if x % 3 == 0 && x % 5 == 0 {
            println!("fizzbuzz")
        } else if x % 3 == 0 {
            println!("fizz")
        } else if x % 5 == 0 {
            println!("buzz")
        } else {
            println!("{}", x)
        }
    }
}

// how to run:
// rustc fizzbuzz2.rs ; ./fizzbuzz2
